src/service/github.oauth.service.ts

```diff
@@ -0,0 +1,71 @@
+import * as express from "express";
+import { Server } from "http";
+import fetch from "node-fetch";
+import { URLSearchParams } from "url";
+import { state } from "../state";
+
+export class GitHubOAuthService {
+  public app: express.Express;
+  public server: Server;
+
+  constructor(public port: number) {
+    this.app = express();
+    this.app.use(express.json(), express.urlencoded({ extended: false }));
+  }
+
+  public async StartProcess() {
+    this.server = this.app.listen(this.port);
+    this.app.get("/callback", async (req, res) => {
+      const params = new URLSearchParams(
+        await (await this.getToken(req.param("code"))).text()
+      );
+
+      res.send("<script>window.close();</script>");
+      this.server.close();
+
+      const token = params.get("access_token");
+      this.saveToken(token);
+
+      const user = await this.getUser(token);
+
+      const gists = await this.getGists(token, user);
+      state.commons.webviewService.OpenGistSelectionpage(gists);
+    });
+  }
+
+  public getToken(code: string) {
+    const params = new URLSearchParams();
+    params.append("client_id", "cfd96460d8b110e2351b");
```

Comment:
is there any way we can hide this? What are your thoughts

Reply:
As far as I know, there is no way to hide the client id/secret in an extension like this.

Reply:
Alright, then I think we can keep it.